crates/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -137,8 +136,7 @@ impl<'a> ModCollector<'a> {
 
             // Add function to scope/ns of the module
             let result = self.def_collector.def_map.modules[self.module_id.0]
-                .scope
-                .define_func_def(name, func_id);
+                .declare_function(name, func_id);
```

Comment:
I also changed these method names to declare_xxx rather than define_xxx as technically we are not actually defining the items, only declaring them. Defining them would entail adding the FuncMeta and function body as well.